From 84d93cd45a743d6e5b1b3136789b627121bbc745 Mon Sep 17 00:00:00 2001
From: Ahmad Abbas <ahmada@marvell.com>
Date: Tue, 5 Jan 2016 19:30:49 +0200
Subject: [PATCH 33/66] defconfig: compiler: added defconfig support for using
 marvell compiler

edited the defconfigs for using external compilers, in order to compile
buildroot need to add the path and prefix of the compiler.

for example:

BR2_TOOLCHAIN_EXTERNAL_PATH="/path/to/compile"
BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="prefix"

Change-Id: Ife82e7f5930c9787c8032ea7fbe6ae9224078c34
Signed-off-by: Ahmad Abbas <ahmada@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/26460
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
Tested-by: Shadi Ammouri <shadi@marvell.com>
---
 configs/mvebu_armv7_be_defconfig      | 13 ++++++++++---
 configs/mvebu_armv7_tiny_le_defconfig |  6 ++++++
 configs/mvebu_armv8_le_defconfig      |  7 +++++++
 configs/mvebu_armv8_tiny_le_defconfig |  6 ++++++
 4 files changed, 29 insertions(+), 3 deletions(-)

diff --git a/configs/mvebu_armv7_be_defconfig b/configs/mvebu_armv7_be_defconfig
index e884f0d..e883711 100644
--- a/configs/mvebu_armv7_be_defconfig
+++ b/configs/mvebu_armv7_be_defconfig
@@ -3,6 +3,13 @@ BR2_cortex_a9=y
 BR2_ARM_EABIHF=y
 BR2_ARM_FPU_VFPV3=y
 BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
+BR2_TOOLCHAIN_EXTERNAL_PATH="/path/to/compile"
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="$(ARCH)"
+BR2_TOOLCHAIN_EXTERNAL_GCC_4_6=y
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_2=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
+BR2_TOOLCHAIN_EXTERNAL_CXX=y
 BR2_MVEBU_RELEASE_NAME="2016_T1.0.eng_drop_v1"
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
 BR2_ROOTFS_OVERLAY="user_files/"
@@ -18,7 +25,6 @@ BR2_PACKAGE_STRACE=y
 BR2_PACKAGE_STRESS=y
 BR2_PACKAGE_WHETSTONE=y
 BR2_PACKAGE_GREP=y
-BR2_PACKAGE_TAR=y
 BR2_PACKAGE_BTRFS_PROGS=y
 BR2_PACKAGE_CRAMFS=y
 BR2_PACKAGE_E2FSPROGS_DEBUGFS=y
@@ -27,7 +33,6 @@ BR2_PACKAGE_DEVMEM2=y
 BR2_PACKAGE_HDPARM=y
 BR2_PACKAGE_I2C_TOOLS=y
 BR2_PACKAGE_IOSTAT=y
-BR2_PACKAGE_KVMTOOL=y
 BR2_PACKAGE_LM_SENSORS=y
 BR2_PACKAGE_LSHW=y
 BR2_PACKAGE_MDADM=y
@@ -50,7 +55,7 @@ BR2_PACKAGE_LIGHTTPD=y
 BR2_PACKAGE_NCFTP=y
 BR2_PACKAGE_NETCAT=y
 BR2_PACKAGE_RSYNC=y
-BR2_PACKAGE_SAMBA=y
+BR2_PACKAGE_SAMBA4=y
 BR2_PACKAGE_TCPDUMP=y
 BR2_PACKAGE_VSFTPD=y
 BR2_PACKAGE_WGET=y
@@ -61,6 +66,8 @@ BR2_PACKAGE_TIME=y
 BR2_PACKAGE_WHICH=y
 BR2_PACKAGE_FTOP=y
 BR2_PACKAGE_HTOP=y
+BR2_PACKAGE_KVMTOOL=y
+BR2_PACKAGE_TAR=y
 BR2_PACKAGE_LESS=y
 BR2_PACKAGE_VIM=y
 BR2_TARGET_ROOTFS_TAR_GZIP=y
diff --git a/configs/mvebu_armv7_tiny_le_defconfig b/configs/mvebu_armv7_tiny_le_defconfig
index 20d74ae..3f345c7 100644
--- a/configs/mvebu_armv7_tiny_le_defconfig
+++ b/configs/mvebu_armv7_tiny_le_defconfig
@@ -2,6 +2,12 @@ BR2_arm=y
 BR2_cortex_a9=y
 BR2_ARM_FPU_VFPV3=y
 BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
+BR2_TOOLCHAIN_EXTERNAL_PATH="/path/to/compile"
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="$(ARCH)-"
+BR2_TOOLCHAIN_EXTERNAL_GCC_4_6=y
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_2=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
 BR2_MVEBU_RELEASE_NAME="2015_T3.0.eng_drop_v1"
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
 BR2_ROOTFS_OVERLAY="user_files"
diff --git a/configs/mvebu_armv8_le_defconfig b/configs/mvebu_armv8_le_defconfig
index e324c69..0936aff 100644
--- a/configs/mvebu_armv8_le_defconfig
+++ b/configs/mvebu_armv8_le_defconfig
@@ -1,6 +1,13 @@
 BR2_aarch64=y
 BR2_OPTIMIZE_2=y
 BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
+BR2_TOOLCHAIN_EXTERNAL_PATH="/path/to/compile"
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="$(ARCH)-"
+BR2_TOOLCHAIN_EXTERNAL_GCC_5=y
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_18=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
+BR2_TOOLCHAIN_EXTERNAL_CXX=y
 BR2_MVEBU_RELEASE_NAME="2016_T1.0.eng_drop_v1"
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
 BR2_ROOTFS_OVERLAY="user_files/"
diff --git a/configs/mvebu_armv8_tiny_le_defconfig b/configs/mvebu_armv8_tiny_le_defconfig
index e9fd5e7..819024b 100644
--- a/configs/mvebu_armv8_tiny_le_defconfig
+++ b/configs/mvebu_armv8_tiny_le_defconfig
@@ -1,5 +1,11 @@
 BR2_aarch64=y
 BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
+BR2_TOOLCHAIN_EXTERNAL_PATH="/path/to/compile"
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_PREFIX="$(ARCH)-"
+BR2_TOOLCHAIN_EXTERNAL_GCC_5=y
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_18=y
+BR2_TOOLCHAIN_EXTERNAL_CUSTOM_GLIBC=y
 BR2_MVEBU_RELEASE_NAME="2016_T1.0.eng_drop_v1"
 BR2_TARGET_GENERIC_GETTY_PORT="ttyS0"
 BR2_ROOTFS_OVERLAY="user_files"
-- 
1.9.1

